index.bs

```diff
@@ -258,6 +262,67 @@ occurred on the [=remote end=].
     - <code>params</code> which defines a mapping containing event data. The
       populated value of this map is the <dfn export for=command>event
       parameters</code>.
+ - A set of [=remote end steps=] which define how to construct the [=event
+   type=] data.
+ - An <dfn export for=event>event trigger</dfn> which is a condition on which
+   the [=remote end steps=] for the event are invoked.
+
+A [=/session=] has a <dfn export for=event>global event set</dfn> which is a set
```

Comment:
That's basically what it's for. I don't know we won't need the pause thing but that seems like a more invasive change in the browser engine and I worry about difficulty of implementation.

Reply:
I've asked about how we do this in Gecko, and it turns out the model we have is closer to what's in the spec; we can get a browsingContextCreated event very early in the lifecycle before anything that would cause other events to happen is run. At that point we run the code to enable things that are in a global event list (and do other things that are necessary to ensure that we don't miss anything on startup e.g. in devtools setup the breakpoints that will be applied). AFAICT we don't have any specific concept of pausing here, so I'd be very reluctant to specify something that requires it. AFAICT the main advantage of the pause model is that you could ask the client which events it wants to enable for that specific browsing context, rather than needing a global list. But with the global list model you can enable events you might want for any browsing context and then explicitly disable the ones that you aren't interested in, without requiring a roundtrip to the client in the context opening code. So overall I'd strongly prefer to keep the model in the spec, but I'd expect blink could implement it on top of the pause machinery if necessary.

Reply:
s/in the spec/in the PR/g

Reply:
Thanks for checking what Gecko does. The mechanism in CDP is the `waitForDebuggerOnStart` flag in [`Target.setAutoAttach`](https://chromedevtools.github.io/devtools-protocol/tot/Target/#method-setAutoAttach). It's a very debugger-oriented model, as the name implies, but I suspect there are things one can do other than enable events. It's notable that Puppeteer *doesn't* use this setting, however, which does cause it to miss events. That's how I found out about it, via @sadym-chromium's investigation of [Event Network.loadingFinished is not sent when a frame is loaded from another domain](https://bugs.chromium.org/p/chromium/issues/detail?id=1060080).

All that being said, I'm not familiar enough with the model to have any strong opinions about its advantages over the "global event set". For common use cases I'd imagine the "global event set" to be easy to reason about, so let's go with that, especially given that it's close to what Gecko does.

If we later find a need to do other stuff early in the lifecycle, then maybe we will define a lower-level thing which "global event set" can the redefined on top of. We shall see.